src/wallet/sqlite.cpp

```diff
@@ -67,6 +71,14 @@ std::unique_ptr<DatabaseBatch> SQLiteDatabase::MakeBatch(const char* mode, bool
     return nullptr;
 }
 
+SQLiteBatch::SQLiteBatch(SQLiteDatabase& database, const char* mode)
+    : m_database(database)
+{
+    m_read_only = (!strchr(mode, '+') && !strchr(mode, 'w'));
+    m_database.AddRef();
+    m_database.Open(mode);
```

Comment:
The purpose of opening the database within a batch is to only open it when the database is going to be used. It is then left open to avoid constantly closing and reopening it.

We also do it to set the mode on the database, but the usefulness of that is arguable but we can deal with that in a followup.